### PR TITLE
Show view api button even if there are no named api routes in the app

### LINKED ISSUE #2160: Show "view API" button with a helpful message even with no named API routes
It has been pretty confusing for our users when they see some Gradio demos with the "view API" link and others without it. The reason this happens is that when there are no named API routes, we do not show the "view API" link. This is the case in Blocks (by default if there is no `api_name` attached to any event listener), while in Interface, there is always at least one named API.

To avoid this confusion, I would propose that we always show the "view API" link, but if there is no named API then clicking on the "view API" link includes a message that says "this demo has no named API routes" and link to the documentation for named API routes.

Related discussion: https://huggingface.slack.com/archives/C02QNREU24S/p1662068541775709

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #2160

### To test

- [x] Case of no named api routes: run `demo/calculator_blocks/run.py`

Should see the "To add api docs" page

image

- [x] Case of named api routes: run modify `demo/calculator_blocks/run.py`   and add an api_name to `submit_btn.click` 

Should see the actual api docs

image

- [x] Case of suppressing the api page when there are named api routes: Set `show_api` to False in the above modified `calculator_blocks` demo

image

You can also do this with an interface demo like `calculator`

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ui/packages/app/src/ApiDocs.svelte [hunk 1]
```diff
@@ -16,99 +16,121 @@
...
 					</div>
-				</div>
-			{/if}
-		{/each}
-	</div>
+				{/if}
+			{/each}
+		</div>
+	{:else}
+		<h2 class="text-3xl text-center mb-6">
```
Comment (freddyaboulton) [inline ui/packages/app/src/ApiDocs.svelte:117]: Big diff but I just added an if else, where the else clause is the case of no named api routes.

File: ui/packages/app/src/ApiDocs.svelte [hunk 2]
```diff
@@ -16,99 +16,121 @@
...
+			There are no named API Routes for
+			<span class="italic text-amber-500">
+				{root}
+			</span>
+		</h2>
+		<div>
+			To expose an api endpoint of your app in the documentation, set the <span
```
Comment (freddyaboulton) [inline ui/packages/app/src/ApiDocs.svelte:124]: Happy to change the text/styling as well

**Comment by freddyaboulton** [comment]
Thanks a really good point @aliabid94 ! 

I wanted to target the "empty state" of the api docs at developers since we get lots of questions from developers asking why the api docs are "missing" from their app. 

That being said, maybe it will be better if developers can disable the documentation via a parameter of the Blocks `launch` method. That way, when they see the "empty state" api docs page, they can decide to either name some of the end points or disable the documentation entirely.

**Comment by abidlabs** [comment]
Agreed on adding a `show_api` parameter to `launch()` to allow developers to hide this page if they want. 

For the message itself, I think we should point developers to the `api_name` parameter in the Guides: https://gradio.app/sharing_your_app/#api-page

So I would suggest the message to read:

--

To expose an API endpoint of your app in these API docs, set the _api_name_ parameter of the event listener. For more information, see the ["API Page" section in the Guides](https://gradio.app/sharing_your_app/#api-page). To hide this API page, set  to _show_api=False_ in the `.launch()` method.

File: ui/packages/app/src/Blocks.svelte [hunk 3]
```diff
@@ -35,6 +35,7 @@
 	export let target: HTMLElement;
 	export let id: number = 0;
 	export let autoscroll: boolean = false;
+	export let show_api: boolean = true;
```
Comment (freddyaboulton) [inline ui/packages/app/src/Blocks.svelte:38]: I think this is backwards compatible. I tested by manually removing `show_api` from the config and this parameter is still set to true. I think that covers us in the case you try to load a config from a demo created in a previous version that doesn't have `show_api`. 

Does this sound right to you @aliabid94 ?

**Comment by freddyaboulton** [comment]
@aliabid94 @abidlabs I added the `show_api` parameter to the Blocks.launch method and modified the text for the "no named routes" page as suggested.

This should be good for another look!

File: ui/packages/app/src/Blocks.svelte (hunk 3 above)
Comment (aliabid94) [reply ui/packages/app/src/Blocks.svelte:38]: sounds good to me

**Review by aliabid94 — APPROVED** [review]
LGTM

**Comment by abidlabs** [comment]
Same LGTM!

**Pull request merged**